Add PEP 561 support

It would be useful to be able to use the type hints defined in this library with mypy. This library needs to add support for PEP 561 for this to work.

@untitaker consider the following Dockerfile:

```
FROM python
RUN pip install mypy sentry-sdk
RUN echo 'import sentry_sdk\nsentry_sdk.init("a")' > a.py
RUN mypy --strict a.py
```

Here's what happens when you run `docker build .`:

```
Sending build context to Docker daemon  2.048kB
Step 1/4 : FROM python
 ---> d6a7b0694364
Step 2/4 : RUN pip install mypy sentry-sdk
 ---> Using cache
 ---> 2e54c901f49a
Step 3/4 : RUN echo 'import sentry_sdk\nsentry_sdk.init("a")' > a.py
 ---> Running in f5b0ac401a61
Removing intermediate container f5b0ac401a61
 ---> 33e2ee6286de
Step 4/4 : RUN mypy --strict a.py
 ---> Running in edd714b6961b
a.py:2: error: Module has no attribute "init"
Found 1 error in 1 file (checked 1 source file)
The command '/bin/sh -c mypy --strict a.py' returned a non-zero code: 1
```

--- a/sentry_sdk/__init__.py
+++ b/sentry_sdk/__init__.py
@@ -4,11 +4,10 @@ from sentry_sdk.transport import Transport, HttpTransport
 from sentry_sdk.client import Client
 
 from sentry_sdk.api import *  # noqa
-from sentry_sdk.api import __all__ as api_all
 
 from sentry_sdk.consts import VERSION  # noqa
 
-__all__ = api_all + [  # noqa
+__all__ = [  # noqa
     "Hub",
     "Scope",
     "Client",
@@ -16,6 +15,22 @@ __all__ = api_all + [  # noqa
     "HttpTransport",
     "init",
     "integrations",
+    # From sentry_sdk.api
+    "capture_event",
+    "capture_message",
+    "capture_exception",
+    "add_breadcrumb",
+    "configure_scope",
+    "push_scope",
+    "flush",
+    "last_event_id",
+    "start_span",
+    "start_transaction",
+    "set_tag",
+    "set_context",
+    "set_extra",
+    "set_user",
+    "set_level",
 ]
 
 # Initialize the debug support after everything is loaded

--- a/sentry_sdk/api.py
+++ b/sentry_sdk/api.py
@@ -27,6 +27,7 @@ else:
         return x
 
 
+# When changing this, update __all__ in __init__.py too
 __all__ = [
     "capture_event",
     "capture_message",

